### PR TITLE
Cross browser (IE) fix for the word limiter

### PR DESCRIPTION
**Pull request by hammerfunk**

### CODE REVIEW

File: app/assets/javascripts/frontend/text-character-count.js [hunk 1]
```diff
@@ -67,7 +100,7 @@ $.fn.charcount = function() {
 
     if (characterOver(counter, textInput)) {
       // hard limit to word count using maxlength
-      if (((typeof(textInput.attr("maxlength")) !== typeof(undefined)) && textInput.attr("maxlength") !== false) == false) {
+      if (((typeof(textInput.attr("maxlength")) !== typeof(undefined)) && textInput.attr("maxlength") !== false) === false) {
```
Comment (matthewford) [inline app/assets/javascripts/frontend/text-character-count.js]: Line is too long.

File: app/assets/javascripts/frontend/text-character-count.js [hunk 2]
```diff
@@ -75,21 +108,68 @@ $.fn.charcount = function() {
     } else {
       textInput.removeAttr("maxlength");
     }
-  }
+
+    $("textarea[maxlength]").each(function() {
+      var text_events = $._data(this, "events");
```
Comment (matthewford) [inline app/assets/javascripts/frontend/text-character-count.js]: Identifier 'text_events' is not in camel case.

File: app/assets/javascripts/frontend/text-character-count.js [hunk 3]
```diff
@@ -75,21 +108,68 @@ $.fn.charcount = function() {
...
+      if (text_events) {
```
Comment (matthewford) [inline app/assets/javascripts/frontend/text-character-count.js]: Identifier 'text_events' is not in camel case.

File: app/assets/javascripts/frontend/text-character-count.js [hunk 4]
```diff
@@ -75,21 +108,68 @@ $.fn.charcount = function() {
...
+        if (!("keydown" in text_events)) {
```
Comment (matthewford) [inline app/assets/javascripts/frontend/text-character-count.js]: Identifier 'text_events' is not in camel case.

File: app/assets/javascripts/frontend/text-character-count.js [hunk 5]
```diff
@@ -75,21 +108,68 @@ $.fn.charcount = function() {
...
+    var textInput = $(this);
 
-    if (originalText.length > 0) {
-      e.preventDefault();
+    if ((e.originalEvent || e).clipboardData) {
+      alert(1);
+      var oldText = textInput.val();
+      var copyText = ((e.originalEvent || e).clipboardData.getData("text/plain"));
+
+      if (!copyText.length > 0) {
```
Comment (matthewford) [inline app/assets/javascripts/frontend/text-character-count.js]: Confusing use of '!'.

File: app/assets/javascripts/frontend/text-character-count.js [hunk 6]
```diff
@@ -75,21 +108,68 @@ $.fn.charcount = function() {
...
-  }
+
+    $("textarea[maxlength]").each(function() {
+      var text_events = $._data(this, "events");
+      if (text_events) {
+        if (!("keydown" in text_events)) {
+          $(this).on("keydown keyup", function (e) {
+            $(this).addClass("char-over-cut");
+            while ($(this).hasClass("char-over-cut")) {
+              Countable.once(this, cutOverChar);
```
Comment (matthewford) [inline app/assets/javascripts/frontend/text-character-count.js]: 'Countable' is not defined.

File: app/assets/javascripts/frontend/text-character-count.js [hunk 7]
```diff
@@ -75,21 +108,68 @@ $.fn.charcount = function() {
...
   // Maxlength for pasting text
-  this.bind("paste", function(e){
-    var originalText = ((e.originalEvent || e).clipboardData.getData("text/plain"));
+  $(this).on("paste", function (e) {
+    var textInput = $(this);
 
-    if (originalText.length > 0) {
-      e.preventDefault();
+    if ((e.originalEvent || e).clipboardData) {
+      alert(1);
```
Comment (matthewford) [inline app/assets/javascripts/frontend/text-character-count.js]: 'alert' is not defined.

File: app/assets/javascripts/frontend/text-character-count.js [hunk 8]
```diff
@@ -75,21 +108,68 @@ $.fn.charcount = function() {
...
+      var oldText = textInput.val();
+      var copyText = ((e.originalEvent || e).clipboardData.getData("text/plain"));
+
+      if (!copyText.length > 0) {
+        e.preventDefault();
+
+        textInput.after("<div id='oldText'>"+oldText+"</div>");
+        Countable.once($("#oldText")[0], countText);
```
Comment (matthewford) [inline app/assets/javascripts/frontend/text-character-count.js]: 'Countable' is not defined.

File: app/assets/javascripts/frontend/text-character-count.js [hunk 9]
```diff
@@ -75,21 +108,68 @@ $.fn.charcount = function() {
...
+        textInput.after("<div id='copyText'>"+copyText+"</div>");
+        Countable.once($("#copyText")[0], countText);
```
Comment (matthewford) [inline app/assets/javascripts/frontend/text-character-count.js]: 'Countable' is not defined.

File: app/assets/javascripts/frontend/text-character-count.js [hunk 10]
```diff
@@ -75,21 +108,68 @@ $.fn.charcount = function() {
...
+        if (newCopytext) {
+          textInput.val(textInput.val() + newCopytext);
           Countable.once(this, counting);
         }
       }
+    } else {
+      setTimeout(function() {
+        textInput.addClass("char-over-cut");
+        while (textInput.hasClass("char-over-cut")) {
+          Countable.once(textInput[0], cutOverChar);
```
Comment (matthewford) [inline app/assets/javascripts/frontend/text-character-count.js]: 'Countable' is not defined.

File: app/assets/javascripts/frontend/text-character-count.js [hunk 11]
```diff
@@ -75,21 +108,68 @@ $.fn.charcount = function() {
     } else {
       textInput.removeAttr("maxlength");
     }
-  }
+
+    $("textarea[maxlength]").each(function() {
+      var text_events = $._data(this, "events");
+      if (text_events) {
+        if (!("keydown" in text_events)) {
+          $(this).on("keydown keyup", function (e) {
```
Comment (matthewford) [inline app/assets/javascripts/frontend/text-character-count.js]: 'e' is defined but never used.

**Comment by RuslanHamidullin** [comment]
Looks good for me :+1:

**Pull request merged**